### PR TITLE
Blazor WASM security updates for 5.0

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #19503

Internal Review Topics ... these are the focus topics of this PR ...

* [Standalone with Auth Lib](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-authentication-library?view=aspnetcore-3.1&branch=pr-en-us-20138)
* [Standalone with Microsoft Accounts](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-microsoft-accounts?view=aspnetcore-3.1&branch=pr-en-us-20138)
* [Standalone with AAD](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-azure-active-directory?view=aspnetcore-3.1&branch=pr-en-us-20138)
* [Standalone with AAD B2C](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-azure-active-directory-b2c?view=aspnetcore-3.1&branch=pr-en-us-20138)

... there are some additional updates to other topics, but they can be somewhat bypassed given that they aren't receiving their full updates here today. They will be the focus of additional PRs in the coming days.

* **All of the updates** for the **standalone** guidance.
* **Part of the updates** for the **hosted** guidance. The AAD and AAD B2C hosted topics will be the focus of a PR tomorrow to finish their 5.0 updates.
* Consistency and correctness nits.
  * Code-fencing "Client" and "Server" when referring to the projects created by the template, which will only be in English.
  * Only using "clear" and "selected" when referring to check boxes per the MS style guidelines.

You'll see that I've temporarily left the NOTE on *If the Azure portal provides the scope URI for the app and the app throws an unhandled exception when it receives a 401 Unauthorized response from the API, ...*. Let's leave that for now. I need to work on that a bit more before I determine if we can safely version it only for 3.x. I'll probably have an answer on it for tomorrow's PR.

I guessed correctly on issue Questions 1-3, so I didn't need to make changes. I already had some general text for `DefaultAccessTokenScopes` with `AdditionalScopesToConsent` added on this PR, but we still don't have anything specific *yet*. If you want to float some content ... a sentence or two ... or more ... on review, let's do that. Otherwise, we can go with this coverage now, and I can ask for further feedback later on it.

**_If possible_**, I'd like to get this merged by EOD so that I'm unblocked to pick up with the AAD-based hosted topics in the morning. Idk if we can pull it off, but I hope we can try. *Sorry for the rush!* 🏃🏃🏃🏃🏃🏃 I don't plan on pinging for additional comments, so I yield to you if you think we should.

### CODE REVIEW

File: aspnetcore/blazor/security/webassembly/standalone-with-authentication-library.md [hunk 1]
```diff
@@ -93,6 +93,8 @@ Authentication support for standalone apps is offered using OpenID Connect (OIDC
 
 ## Access token scopes
 
+The Blazor WebAssembly template automatically configures default scopes for `openid` and `profile`.
```
Comment (captainsafia) [inline aspnetcore/blazor/security/webassembly/standalone-with-authentication-library.md:96]: Only for 3.x?

File: aspnetcore/blazor/security/webassembly/standalone-with-azure-active-directory-b2c.md [hunk 2]
```diff
@@ -67,12 +91,18 @@ dotnet new blazorwasm -au IndividualB2C --aad-b2c-instance "{AAD B2C INSTANCE}"
...
-> In the Azure portal, the app's **Authentication** > **Platform configurations** > **Web** > **Redirect URI** is configured for port 5001 for apps that run on the Kestrel server with default settings.
+> In the Azure portal, the app's platform configuration **Redirect URI** is configured for port 5001 for apps that run on the Kestrel server with default settings.
 >
 > If the app is run on a random IIS Express port, the port for the app can be found in the app's properties in the **Debug** panel.
 >
 > If the port wasn't configured earlier with the app's known port, return to the app's registration in the Azure portal and update the redirect URI with the correct port.
 
+::: moniker range=">= aspnetcore-5.0"
+
+[!INCLUDE[](~/includes/blazor-security/additional-scopes-standalone.md)]
```
Comment (captainsafia) [inline aspnetcore/blazor/security/webassembly/standalone-with-azure-active-directory-b2c.md:102]: AAD and B2C needs different scopes so I think we'll need different versions of these includes.

AAD needs the `User.Read` scope and `B2C` uses the `openid` ones.

Also, the config shouldn't be needed for hosted apps.

**Review by captainsafia — COMMENTED** [review]
Pretty solid overall! Left some comments inline about clarifying the scopes that are set.

File: aspnetcore/blazor/security/webassembly/standalone-with-authentication-library.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/blazor/security/webassembly/standalone-with-authentication-library.md:96]: Both 3.x and 5.x ...

https://github.com/dotnet/aspnetcore/blob/648c15dbe90fb8c113f7c6b4adeb40d9e10494f6/src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs#L33

https://github.com/dotnet/aspnetcore/blob/release/5.0-rc2/src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs#L33

... if I have the 5.0 branch correct. :smile:

Comment (captainsafia) [reply aspnetcore/blazor/security/webassembly/standalone-with-authentication-library.md:96]: Ah, boo. This is the OIDC provider. I was thinking about a different codepath.

File: aspnetcore/blazor/security/webassembly/standalone-with-azure-active-directory-b2c.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/blazor/security/webassembly/standalone-with-azure-active-directory-b2c.md:102]: I only tested standalone with Microsoft Accounts this morning. The scopes are only `openid` and `offline_access` for that scenario. I incorrectly assumed that the AAD and AAD B2C scenarios would match. I'll get this resolved on the next commit.

> Also, the config shouldn't be needed for hosted apps.

Do you mean for the *Server* app of the hosted solution? I agree, but let's focus on the two hosted scenarios on another PR, perhaps tomorrow.

Comment (guardrex) [reply aspnetcore/blazor/security/webassembly/standalone-with-azure-active-directory-b2c.md:102]: Ah! ... yes ... I see what you mean now. Good :eye:. I really should have taken a little more time with it this morning.

**Comment by guardrex** [comment]
@captainsafia ... Ok ... thx for that ... I wish I would've drilled down a bit more this morning.

We now have `-AAD` (for AAD) and `-nonAAD` (for MS Accounts and B2C) INCLUDES for both standalone and hosted default scopes.

> the config shouldn't be needed for hosted apps

I thought that the *Client* app of a hosted solution would be configured like a standalone app ... at least to avoid this new error for an app that doesn't specify default scopes. That's a *wrong guess*? I would find out the hard way in the morning anyway ... I'll be working with the hosted scenarios tomorrow.

**Comment by guardrex** [comment]
@captainsafia ... I'll go ahead and merge now so that I can pick up with the next round of updates. If there's more to update from this PR, I'll place the changes on the next one.

**Pull request merged**